hook-producer/src/handlers/mod.rs

```diff
@@ -1,15 +1,164 @@
+                Request::builder()
+                    .method(http::Method::POST)
+                    .uri("/webhook")
+                    .header(http::header::CONTENT_TYPE, "application/json")
+                    .body(Body::from(
+                        serde_json::to_string(&WebhookJobParameters {
+                            headers,
+                            method: HttpMethod::POST,
+                            url: "http://example.com/".to_owned(),
+                            body: r#"{"a": "b"}"#.to_owned(),
+                        })
+                        .unwrap(),
+                    ))
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let body = response.into_body().collect().await.unwrap().to_bytes();
+        assert_eq!(&body[..], b"{}");
+    }
+
+    #[tokio::test]
+    async fn webhook_bad_url() {
+        let pg_queue = PgQueue::new(
+            "test_index",
+            "job_queue",
+            "postgres://posthog:posthog@localhost:15432/test_database",
+            RetryPolicy::default(),
+        )
+        .await
+        .expect("failed to construct pg_queue");
+
+        let app = app(pg_queue, None);
+
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .method(http::Method::POST)
+                    .uri("/webhook")
+                    .header(http::header::CONTENT_TYPE, "application/json")
+                    .body(Body::from(
+                        serde_json::to_string(&WebhookJobParameters {
+                            headers: collections::HashMap::new(),
+                            method: HttpMethod::POST,
+                            url: "invalid".to_owned(),
+                            body: r#"{"a": "b"}"#.to_owned(),
+                        })
+                        .unwrap(),
+                    ))
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::BAD_REQUEST);
+    }
+
+    #[tokio::test]
+    async fn webhook_bad_payload() {
+        let pg_queue = PgQueue::new(
+            "test_index",
+            "job_queue",
+            "postgres://posthog:posthog@localhost:15432/test_database",
+            RetryPolicy::default(),
+        )
+        .await
+        .expect("failed to construct pg_queue");
+
+        let app = app(pg_queue, None);
+
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .method(http::Method::POST)
+                    .uri("/webhook")
+                    .header(http::header::CONTENT_TYPE, "application/json")
+                    .body("{}".to_owned())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+    }
```

Comment:
I can definitely add more test coverage given time, I just got to the basics today.